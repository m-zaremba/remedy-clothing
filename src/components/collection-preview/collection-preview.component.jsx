import React from 'react';
import './collection-preview.styles.scss';
import PropTypes, { object } from 'prop-types';
import CollectionItem from '../collection-item/collection-item.component';

const CollectionPreview = ({ title, items }) => {
  return (
    <div className="collection-preview">
      <h1 className="title">{title.toUpperCase()}</h1>
      <div className="preview">
        {items
          .filter((item, idx) => idx < 4)
          .map((item) => (
            <CollectionItem key={item.id} item={item} />
          ))}
      </div>
    </div>
  );
};

CollectionPreview.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(object).isRequired,
};

export default CollectionPreview;
