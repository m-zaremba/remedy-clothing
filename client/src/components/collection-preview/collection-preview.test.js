import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './collection-preview.component';

describe('testing CollectionPreview component', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const mockRouteName = 'mens';

  beforeEach(() => {
    mockMatch = {
      path: '/shop',
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: 'mens',
      items: [],
    };

    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  it('should render CollectionPreview component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push with the right string when Title clicked', () => {
    wrapper.find('TitleButton').simulate('click');

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.path}/${mockRouteName}`);
  });
});
