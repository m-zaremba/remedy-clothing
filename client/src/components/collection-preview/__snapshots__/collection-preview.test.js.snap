// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CollectionPreview component should render CollectionPreview component 1`] = `
<styled.div>
  <TitleButton
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    MENS
  </TitleButton>
  <styled.div />
</styled.div>
`;
